packages/openid4vc/src/openid4vc-holder/OpenId4VcHolderApi.ts

```diff
@@ -146,11 +146,12 @@ export class OpenId4VcHolderApi {
    * @param options.code The authorization code obtained via the authorization request URI
    */
   public async requestToken(options: OpenId4VciRequestTokenOptions): Promise<OpenId4VciRequestTokenResponse> {
-    const { access_token: accessToken, c_nonce: cNonce } = await this.openId4VciHolderService.requestAccessToken(
-      this.agentContext,
-      options
-    )
-    return { accessToken, cNonce }
+    const {
+      access_token: accessToken,
+      c_nonce: cNonce,
+      dpop,
+    } = await this.openId4VciHolderService.requestAccessToken(this.agentContext, options)
```

Comment:
We can change the service interface just fine, so let's change the service return value to:
```
{
  accessToken,
  cNonce,
  dpop,
  response // should be cloned response (using .clone)
}
```

Reply:
? what should the response contain? the whole response?

Reply:
Yes it would be a start at solving https://github.com/openwallet-foundation/credo-ts/issues/1965, which was requested. Returning the response means better extensibility as you can extract the headers, etc..

Reply:
In the PR where I implemented the tokenRequestfunctionality, I had that behaviour, and you made me change it.

Reply:
Okay then i misunderstood the previous implementation. What i don't want is requiring the whole access token response as input to the request credentials method.

I want to have separation between the required fields for credo to work, and additional params you may need for extension. And as I said in the previous PR, i'm not a fan of mixing the successBody with extra params we extracted from e.g. the header.